### DISCUSSION TITLE
Webpack Compiling Slow

### DISCUSSION THREAD

**Discussion by seanmpage** [krjo/shopify-webpack-dev-workflow#11] [Q&A]
Hey!
Really appreciate this project. I was wondering if it was normal for webpack to update every file when you change a single file. I ask because it can take some time when making changes and then waiting for it to update the theme.

**Comment by krjo** [comment]
The webpack watch command should only rebuild the entry files that your current working file is imported in. For example: if you are working in a JS component file that is imported in product.js and collection.js entry files, when you make a change and save the component file, webpack will rebuild product.js and collection.js and both will be uploaded to Shopify. However, with the recent update to Theme Kit that introduced checksums, I did notice that if I switch back and forth between git branches, Theme Kit notices differences between the working branch and the remote and will upload a lot of files when I switch to a different branch. Pretty annoying but there's currently no theme kit flag to disable the checksums. If you're experience something different from both of those, can you elaborate or explain how to reproduce?

**Reply by krjo** [reply]
To add to this...if you run npm run build or npm run deploy, that runs a production build which updates every js/css file in dist/. If you then run npm start to begin dev work again, the new checksum feature in theme kit will recognize that every file has been rebuilt and it will run through and upload every file again. If shopify ever releases a flag to disable checksums, I will probably add that because to me it's more of a pain than a benefit.
